### ISSUE TITLE
Upgrading Kong unexpectedly deletes CRDs and CR resources

### ISSUE THREAD

**Issue by object88** [Kong/charts#610]
Our development cluster has a Kong deployment that has a long upgrade history.  Initially, CRDs were deployed as a part of the chart.  Later in time, in the `values.yaml` file, we set:
```
ingressController:
  installCRDs: false
```
We have since been manually updating the CRD versions.

However, we just upgraded from Kong chart 2.6.2 to 2.9.0, and the Kong CRDs were deleted, as were the CR resources (tcp ingress, plugins, etc.)

AFAIKT, it is related to [this change](https://github.com/Kong/charts/commit/c355905c33a7863148ac80274242cd405c0c933f#diff-4dc49f7de362f78d7f141247aa1833dc2be43ded9651325b69001a2e1180f85aR2-R4).  I don't see anything in `CHANGELOG.md` or `UPGRADE.md` that warns of this behavior.

**Comment by object88** [comment]
From what I can tell, the behavior has changed so that, in the past, if the Helm chart _had_ been installed, then it would be retained, regardless of how `installCRDs` was set.  The current behavior is such that the value is taken literally, so it will *remove* CRDs if they had been present.

This is a breaking change.

**Comment by rainest** [comment]
This actually breaking requires a fairly specific sequence of events. We didn't expect there to be configurations where `installCRDs: false` was explicitly set in values.yaml despite the CRDs having been installed with it set to `true`, only configurations where `installCRDs` was omitted entirely and therefore default `false` after we removed it from the default values.yaml.

In the default values.yaml, `installCRDs` has [only ever been set to true](https://github.com/Kong/charts/blame/kong-1.15.1/charts/kong/values.yaml#L361). [Changes included in 2.0.0](https://github.com/Kong/charts/pull/305/files#diff-e128b1c91d5e0dffa1146ca290e9ed46f8f381e81808192d79f7c8dbbc8bf637L361) essentially removed it altogether, though templates would still honor it if it was present and set to `true`. If it was absent or set to `false`, the chart relied on the presence of release annotations on CRDs to automatically detect whether it owned those resources.

For 2.8, we began honoring `false` again, because permissions to check the CRDs aren't guaranteed to be available, with the expectation that no installs where the release managed CRDs would have explicitly set that key to `false`. While the original implementation could not distinguish between an explicit `false` value and Helm's implied `false` when a key is simply not present, the current one can: omitting the key will still use the automatic detection behavior, while including it will use the key's value.

There were, however, several pieces of documentation and example values from prior to the 2.0 chart release that did include `installCRDs: false` because we had started supporting Helm 3 and recommended using Helm 3 CRD management. These were never updated after we effectively removed `installCRDs`. What I expect happened here was something along the lines of:

1. The chart was initially installed prior to 2.0, with the `installCRDs: true` default.
2. At some point after upgrading to 2.0, the values.yaml used was amended to include an explicit `installCRDs: false`. The templates ignored this setting.
3. After installing 2.8+ the chart did honor the explicit `false`

I've since removed all mention of setting that key to `false` outside its current intended use case from documentation and examples and have proposed an additional changelog warning.
